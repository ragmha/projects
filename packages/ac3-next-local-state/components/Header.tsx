import TodoTextInput from "./TodoTextInput"

type HeaderProps = {
  addTodo: (text: string) => void
}
export const Header = (props: HeaderProps) => {
  return (
    <header className="header">
      <h1>TODOS</h1>
      <TodoTextInput
        newTodo
        onSave={(text) => {
          if (text.length !== 0) {
            props.addTodo(text)
          }
        }}
        placeholder="What needs to be done?"
      />
    </header>
  )
}

export default Header
