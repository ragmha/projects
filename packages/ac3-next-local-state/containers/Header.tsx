import HeaderComponent from "components/Header"
import { TodoMutations } from "operations/mutations"

export const Header = () => {
  return <HeaderComponent addTodo={TodoMutations.addTodo} />
}

export default Header
