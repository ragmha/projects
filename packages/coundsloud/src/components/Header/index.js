import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import * as actions from '../../actions/index';

const Logo = () => (
  <div>
    <Link to="/">
      <h1>SoundCloud</h1>
    </Link>
  </div>
);

const Login = ({ onLogin }) => <Link onClick={onLogin} to="/dashboard" />;
const Logout = ({ onLogout }) => <Link onClick={onLogout} to="/" />;

const SessionAction = ({ currentUser, onLogin, onLogOut }) => (
  <div>
    {currentUser ? (
      <Logout onLogout={onLogOut} />
    ) : (
      <Logout onLogin={onLogin} />
    )}
  </div>
);

const Header = ({ currentUser, onLogin, onLogout }) => (
  <div className="header">
    <div className="header-content">
      <Logo />
      <SessionAction
        currentUser={currentUser}
        onLogin={onLogin}
        onLogout={onLogout}
      />
    </div>
  </div>
);

const mapStateToProps = state => ({
  currentUser: state.session.user,
});

const mapDispatchToProps = dispatch => ({
  onLogin: bindActionCreators(actions.logInSCuser, dispatch),
  onLogOut: bindActionCreators(actions.logOutSCuser, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);
