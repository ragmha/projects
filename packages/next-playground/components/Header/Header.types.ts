export type HeaderLinkProps = {
  active?: boolean
}

export type HeaderProps = HeaderLinkProps
