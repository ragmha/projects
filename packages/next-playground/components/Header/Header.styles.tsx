import styled, { css } from 'styled-components'
import { HeaderLinkProps } from './Header.types'

export const StyledHeader = styled.header`
  margin-bottom: 25px;
`

export const StyledHeaderLink = styled.a<HeaderLinkProps>`
  font-size: 14px;
  margin-right: 15px;
  text-decoration: none;
  cursor: pointer;

  ${({ active }) =>
    active &&
    css`
      text-decoration: underline;
    `}
`
