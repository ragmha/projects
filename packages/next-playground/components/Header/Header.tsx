import { FC } from 'react'
import { useRouter } from 'next/router'
import Link from 'next/link'

import { StyledHeader, StyledHeaderLink } from './Header.styles'
import { HeaderProps } from './Header.types'

export const Header: FC<HeaderProps> = () => {
  const { pathname } = useRouter()

  return (
    <StyledHeader>
      <Link href="/">
        <StyledHeaderLink active={pathname === '/'}>Home</StyledHeaderLink>
      </Link>
      <Link href="/about">
        <StyledHeaderLink active={pathname === '/about'}>
          About
        </StyledHeaderLink>
      </Link>
      <Link href="/client-only">
        <StyledHeaderLink active={pathname === '/client-only'}>
          Client-Only
        </StyledHeaderLink>
      </Link>
    </StyledHeader>
  )
}

export default Header
