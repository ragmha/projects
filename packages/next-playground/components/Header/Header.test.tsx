import Header from './Header'
import { render } from '@utils/test'

describe('Header', () => {
  it('should match the snapshot', () => {
    const { container } = render(<Header active={false} />, {
      router: { pathname: '/' },
    })
    expect(container).toMatchSnapshot()
  })
})
