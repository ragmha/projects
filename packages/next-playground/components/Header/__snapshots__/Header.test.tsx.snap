// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
<div>
  <header
    class="sc-AxjAm hnzZeL"
  >
    <a
      class="sc-AxirZ gVDPQE"
    >
      Home
    </a>
    <a
      class="sc-AxirZ jIokEl"
    >
      About
    </a>
    <a
      class="sc-AxirZ jIokEl"
    >
      Client-only
    </a>
  </header>
</div>
`;
